ndcube/ndcube.py

```diff
@@ -629,6 +630,74 @@ def explode_along_axis(self, axis):
         # Creating a new NDCubeSequence with the result_cubes and common axis as axis
         return NDCubeSequence(result_cubes, meta=self.meta)
 
+    def reproject(self, target_wcs, shape_out=None, order='bilinear', output_array=None, return_footprint=False):
+        """
+        Reprojects this NDCube to the coordinates described by another WCS object.
+
+        Parameters
+        ----------
+        target_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
```

Comment:
I think we should also allow a dict / `fits.Header` instance here as well and turn it into a WCS?

Reply:
```
isinstance(target_wcs, collections.abc.Mapping):
    target_wcs = astropy.wcs.WCS(target_wcs)
```